File: backend/app/api/routes/roi.py
from fastapi import APIRouter

from app.models.roi import RoiRequest, RoiResponse
from app.services.roi_service import calculate_roi

router = APIRouter(prefix="/roi", tags=["ROI"])


@router.post("", response_model=RoiResponse)
def post_roi(payload: RoiRequest) -> RoiResponse:
    return calculate_roi(payload)

File: frontend/src/services/roi.ts
import api from "./api";
import type { RoiRequest, RoiResponse } from "@/types/roi";

export async function calculateRoi(payload: RoiRequest): Promise<RoiResponse> {
  const { data } = await api.post<RoiResponse>("/roi", payload);
  return data;
}
